### ISSUE TITLE
Pass in logger for subscriber middleware

### ISSUE THREAD

**Issue by rizowski** [LeisureLink/magicbus#25]
middleware contract uses 
```javascript
function (message, actions){}
```
We should add
```javascript
function (message,actions,logger){}
```

**Comment by twindagger** [comment]
Not sure i understand, you want to override the logger for just consumption here? Is this still valid after 2.0?

**Comment by rizowski** [comment]
Is there still middleware for magicbus? This issue related to magicbus middleware contracts 
https://github.com/LeisureLink/magicbus-middleware-authentic/blob/master/src/publisher-middleware.js#L5-L11
I poorly described that I wanted magicbus to pass its logger to the middleware so that way the middleware could use the same logger.

**Comment by twindagger** [comment]
Implemented in 2.2.0
